src/query_contract_ui/eigenlayer.c

```diff
@@ -0,0 +1,227 @@
+/*******************************************************************************
+ *
+ * ██╗  ██╗██╗██╗     ███╗   ██╗
+ * ██║ ██╔╝██║██║     ████╗  ██║
+ * █████╔╝ ██║██║     ██╔██╗ ██║
+ * ██╔═██╗ ██║██║     ██║╚██╗██║
+ * ██║  ██╗██║███████╗██║ ╚████║
+ * ╚═╝  ╚═╝╚═╝╚══════╝╚═╝  ╚═══╝
+ *
+ * Kiln Ethereum Ledger App
+ * (c) 2022-2024 Kiln
+ *
+ * contact@kiln.fi
+ ********************************************************************************/
+
+#include "query_contract_ui.h"
+
+/**
+ * @brief UI for depositing into an EigenLayer strategy
+ *
+ * @param msg: message containing the parameter
+ * @param context: context with provide_parameter data
+ *
+ */
+bool deposit_into_strategy_ui(ethQueryContractUI_t *msg, context_t *context) {
+    bool ret = false;
+    lr_deposit_t *params = &context->param_data.lr_deposit;
+
+    switch (msg->screenIndex) {
+        case 0:
+            strlcpy(msg->title, "Liquid Restaking", msg->titleLength);
+            strlcpy(msg->msg, "Deposit In Strategy", msg->msgLength);
+            ret = true;
+            break;
+        case 1:
+            strlcpy(msg->title, "Strategy", msg->titleLength);
+            if (params->strategy_to_display == UNKNOW_LR_STRATEGY ||
+                params->strategy_to_display >= LR_STRATEGIES_COUNT) {
+                strlcpy(msg->msg, "UNKNOWN", msg->msgLength);
+            } else {
+                strlcpy(msg->msg, lr_tickers[params->strategy_to_display], MAX_TICKER_LEN);
+            }
+            ret = true;
+            break;
+        case 2:
+            strlcpy(msg->title, "Amount", msg->titleLength);
+            amountToString(params->erc20_amount_to_display,
+                           sizeof(params->erc20_amount_to_display),
+                           ERC20_DECIMALS,
+                           params->erc20_to_display == UNKNOW_LR_STRATEGY ||
```

Comment:
UNKNOW_LR_STRATEGY is also used in the erc20 context ?
maybe it can lead do confusion wdyt ?